backend/nodes/properties/inputs/numeric_inputs.py

```diff
@@ -0,0 +1,144 @@
+from .base_input import BaseInput
+
+
+class NumberInput(BaseInput):
+    """Input a number"""
+
+    def __init__(
+        self,
+        label: str,
+        default=0.0,
+        minimum=0,
+        maximum=None,
+        step=1,
+        optional=False,
+        number_type="any",
+    ):
+        super().__init__(f"number::{number_type}", label)
+        self.default = default
+        self.minimum = minimum
+        self.maximum = maximum
+        self.step = step
+        self.optional = optional
+
+    def toDict(self):
+        return {
+            "type": self.input_type,
+            "label": self.label,
+            "min": self.minimum,
+            "max": self.maximum,
+            "def": self.default,
+            "step": self.step,
+            "hasHandle": True,
+            "optional": self.optional,
+        }
+
+    def enforce(self, value):
+        return max(float(self.minimum), float(value))
```

Comment:
Could you enforce the maximum (if present) too?

Other numeric inputs also seem to have this problem. Could you fix it for all of them?

Reply:
Good point since maximum can be overridden. Maybe I should default those to -Infinity and Infinity though instead of None? I'll have to test how that works with min/max